partition/roles/bmc-proxy/tasks/main.yaml

```diff
@@ -0,0 +1,104 @@
+---
+- name: Check mandatory variables for this role are set
+  assert:
+    fail_msg: "not all mandatory variables given, check role documentation"
+    quiet: yes
+    that:
+      - bmc_proxy_image_name is defined
+      - bmc_proxy_image_tag is defined
+      - bmc_reverse_proxy_image_name is defined
+      - bmc_reverse_proxy_image_tag is defined
+
+- name: Obtain docker bridge ip
+  command: docker network inspect bridge
+  register: docker_inspect_result
+  when: bmc_proxy_docker_bridge_ip is not none
+
+- name: Set docker bridge ip
+  set_fact:
+    bmc_proxy_docker_bridge_ip: "{{ (docker_inspect_result | from_json)[0].get("IPAM").get("Config")[0].get("Gateway") }}"
+  when: bmc_proxy_docker_bridge_ip is not none
+
+- name: create bmc-proxy host directory
+  file:
+    dest: "{{ bmc_proxy_host_dir_path }}"
+    state: directory
+    owner: "{{ bmc_proxy_cert_owner }}"
+    group: "{{ bmc_proxy_cert_group }}"
+    mode: 0755
+
+- name: copy bmc-reverse-proxy nginx.conf
+  copy:
+    src: nginx.conf
+    dest: "{{ bmc_proxy_host_dir_path }}/"
+    mode: 0644
+    owner: "{{ bmc_proxy_cert_owner }}"
+    group: "{{ bmc_proxy_cert_group }}"
+
+- name: create bmc-proxy certificates directory
+  file:
+    dest: "{{ bmc_proxy_cert_dir_path }}"
+    state: directory
+    owner: "{{ bmc_proxy_cert_owner }}"
+    group: "{{ bmc_proxy_cert_group }}"
+    mode: 0700
+
+- name: copy ca certificates
+  copy:
+    dest: "{{ bmc_proxy_cert_dir_path }}/ca.pem"
+    content: "{{ metal_api_bmc_proxy_certs_ca_cert }}"
```

Comment:
We should not rely on another role's variable. Please define `bmc_proxy_certs_ca_cert` to make this role fully self-encapsulated. The user must parametrize the deployment accordingly.